ddht/v5_1/client.py

```diff
@@ -396,6 +399,82 @@ async def find_nodes(
 
                 return responses
 
+    @asynccontextmanager
+    async def stream_find_nodes(
+        self,
+        node_id: NodeID,
+        endpoint: Endpoint,
+        distances: Collection[int],
+        *,
+        request_id: Optional[bytes] = None,
+    ) -> AsyncIterator[trio.abc.ReceiveChannel[InboundMessage[FoundNodesMessage]]]:
+        with self.request_tracker.reserve_request_id(node_id, request_id) as request_id:
+            request = AnyOutboundMessage(
+                FindNodeMessage(request_id, tuple(distances)), endpoint, node_id,
+            )
+
+            send_channel, receive_channel = trio.open_memory_channel[
+                InboundMessage[FoundNodesMessage]
+            ](256)
```

Comment:
Given that we don't expect to regularly receive more than maybe 2-3 payloads for a given request in normal situations I would recommend lowering this limit to something like 4.